Sophos/sophos edr/ingest/parser.yml

```
@@ -0,0 +1,83 @@
+name: sophos-edr
+pipeline:
+-   external:
+        name: json.parse-json
+        properties:
+            input_field: '{{original.message}}'
+            output_field: message
+    name: parse_json
+-   external:
+        name: grok.match
+        properties:
+            input_field: '{{parse_json.message.name}}'
+            output_field: message
+            pattern: 'Access was blocked to "%{URL:url}" because of "%{WORD:rulename}".'
+            custom_patterns:
+              URL: '(?:%{URI}|%{URIHOST}|%{URIPATHPARAM}|(?:%{IPORHOST})?:%{POSINT:port})'
+    filter: '{{parse_json.message.type == "Event::Endpoint::WebFilteringBlocked"}}'
+    name: parse_webfiltering
+-   external:
+      name: date.parse
+      properties:
+        input_field: parse_json.message.rt
+        output_field: date
+    name: parse_date
+-   name: set_ecs_fields
+-   name: set_sophos_fields
+
+stages:
+  set_ecs_fields:
+    actions:
+    - set:
+        '@timestamp': '{{parse_date.date}}'
+    - set:
+        event.end: '{{parse_json.message.end}}'
+        event.kind: 'event'
+        event.reason: '{{parse_json.message.name}}'
+        event.code: '{{parse_json.message.type}}'
+        event.type: ['info']
+    - set:
+        user.id: '{{parse_json.message.duid}}'
+        host.hostname: '{{parse_json.message.dhost}}'
+        observer.ip: '{{parse_json.message.source_info.ip}}'
+    - set:
+        source.user.name: '{{parse_json.message.suser.split("\\") | last}}'
+        source.user.domain: '{{parse_json.message.suser.split("\\")[0:-1] | join("\\")}}'
```

```suggestion
        user.name: '{{parse_json.message.suser.split("\\") | last}}'
        user.domain: '{{parse_json.message.suser.split("\\")[0:-1] | join("\\")}}'
```

I suggest that you remove `source.` to simplify parser output since you don't use `destination.`

---

(/!\ this modification will break stuff like smart descriptions)